### ISSUE TITLE
Fix S3264 FP: Event invoked through its invocation list acquired by GetInvocationList()

### ISSUE THREAD

**Issue by csaba-sagi-sonarsource** [SonarSource/sonar-dotnet#3931]
### Description

Rule S3264 raises a false positive when the event is invoked through its invocation list acquired by GetInvocationList().

Source: https://community.sonarsource.com/t/remove-unused-event/35730

### Repro steps

```
        public delegate void SomeDelegate(int a);

        public class ReproForIssue
        {
            public event SomeDelegate ActuallyUsedEvent; // Noncompliant FP

            public void SomeMethod(SomeDelegate usedEvent)
            {
                var invocations = usedEvent.GetInvocationList();

                foreach (var invocation in invocations)
                {
                    invocation.DynamicInvoke(1623);
                }
            }
        }
```

### Expected behavior

S3264  should not raise an issue for this case.

### Actual behavior

False positive is reported.

### Known workarounds

None

### Related information

* C#/VB.NET Plugins version: 8.16

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1929](https://sonarsource.atlassian.net/browse/NET-1929)

[NET-1929]: https://sonarsource.atlassian.net/browse/NET-1929?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
